.github/workflows/doxygen-gh-pages.yml

```diff
@@ -1,15 +1,67 @@
-name: Doxygen GitHub Pages Deploy Action
+name: Run Doxygen and publish to GitHub pages
+
+# Generate the documentation on all merges to main, all pull requests, or by
+# manual workflow dispatch. The Doxygen job can be used as a CI check that all
+# functions are documented. The publish job only runs on new commits to the main
+# branch and actually pushes the generated html to the gh-pages branch (which
+# triggers a GitHub pages deployment). When things are a bit more stable, we can
+# only publish the website on release tags.
 on:
   push:
     branches:
       - main
+  pull_request:
   workflow_dispatch:
+
 jobs:
-  deploy:
+  doxygen:
+    runs-on: macos-latest
+    steps:
+
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        submodules: "true"
+
+    - name: Install Doxygen
+      run: brew install doxygen graphviz
+      shell: bash
+
+    - name: Run Doxygen
+      run: doxygen doc/Doxyfile
+      shell: bash
+
+    # Disabling jekyll ensures pages with underscores work on gh pages.
```

Comment:
For my own clarification: you mean pages whose names `contain_underscores` for example?

Reply:
Yes. We don't have any, but we might. And then it'd be a whole nightmare to debug.